### PR TITLE
Update FromPixelsImageBitmap Pipeline Cache Strategy

### PR DESCRIPTION
**Pull request by shaoboyan**
In previous code, FromPixelsImageBitmap will cache the whole program
when it has been created first time and keep it unchanged.

But preprocessor will change the shader code based on output shape. So
the cache strategy of FromPixelsImageBitmap is not correct.

This CL updated the strategy to use WebGPU pipeline cache system and
check out shape for every invocation.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4499)

### CODE REVIEW

**Comment by shaoboyan** [comment]
@qjia7  PTAL , thanks!

**Comment by shaoboyan** [comment]
@qjia7  PTAL again!

**Comment by qjia7** [comment]
LGTM, thanks.

+@kainino0x @pyu10055 @lina128 review, thanks.

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts [hunk 1]
```diff
@@ -92,25 +92,32 @@ export class FromPixelsProgram implements WebGPUProgram {
   }
 
   setUniform(device: GPUDevice, uniformData: number[]) {
+    // Create the uniform buffer if it is not exists.
+    // The uniform buffer size is stick so we can hold
+    // and reuse it always.
```
Comment (kainino0x) [inline tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:97]:
```suggestion
    // Create the uniform buffer if it does not exist.
    // The uniform buffer size is fixed so we can hold
    // and reuse it always.
```

**Review by kainino0x — APPROVED** [review]

**Comment by shaoboyan** [comment]
@kainino0x  thanks for the change and bots trigger.

**Pull request merged**